sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_settings_client.py

```diff
@@ -61,17 +63,18 @@ async def extract_data(_):
         return AsyncItemPaged(get_next, extract_data)
 
     @distributed_trace_async
-    async def update_setting(self, name: str, value: str, **kwargs) -> KeyVaultSetting:
+    async def update_setting(self, name: str, value: Union[bool, str], **kwargs) -> KeyVaultSetting:
         """Updates a given account setting with the provided value.
 
         :param str name: The name of the account setting to update.
-        :param str value: The value to set.
+        :param value: The value to set.
+        :type value: bool or str
 
         :returns: The updated account setting, as a :class:`~azure.keyvault.administration.KeyVaultSetting`.
         :rtype: ~azure.keyvault.administration.KeyVaultSetting
         :raises: :class:`~azure.core.exceptions.HttpResponseError`
         """
-        parameters = UpdateSettingsRequest(value=value)
+        parameters = UpdateSettingsRequest(value=str(value).lower())
```

Comment:
Wont this lowercase the value of any string `value` passed in? Are all `setting` values case insensitive?

Reply:
That's a good point! So far I only know of boolean settings, but if there were case-sensitive string settings then this could be an issue. I can add a boolean check before `.lower`ing anything